### PR TITLE
Introduce editorconfig rules.

### PR DESCRIPTION
**Pull request by JimiC**
This PR introduces `editorconfig` ruling. This way we will get rid of the constant changing of spacing in `package.json`.

@robertohuertasm If you want `package.json` spacing as the rest of the `json` files just remove the `[package.json]` rule.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
 {
-  "tslint.enable": true,
```
Comment (JimiC) [inline .vscode/settings.json]: Already a default setting in `vscode`. That's why I removed it.

Comment (robertohuertasm) [inline .vscode/settings.json]: Please, keep it. As I have it disabled in my global settings and I like to manage it per project.

Comment (JimiC) [inline .vscode/settings.json]: Oh, I see.

File: package.json [hunk 2]
```diff
@@ -1,176 +1,176 @@
...
+    "Other",
+    "Themes"
+  ],
+  "keywords": [
+    "icon-theme",
+    "theme",
+    "icons",
+    "vscode-icons"
+  ],
+  "preview": false,
```
Comment (JimiC) [inline package.json]: Why? What's the purpose of it in this project? Are still in preview (beta)?
It was disabled in a previous commit.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
+indent_style = space
+indent_size = 4
```
Comment (JimiC) [inline .editorconfig]: Actually we aren't. But if so it's no problem.

File: package.json (hunk 2 above)
Comment (robertohuertasm) [inline package.json]: I asked to keep it in preview in the previous PR but somehow I missed it and this change got in.

Comment (JimiC) [inline package.json]: Can you explain why we are in preview?

File: .editorconfig (hunk 3 above)
Comment (robertohuertasm) [inline .editorconfig]: If not, the idea was to use 2
```
"editor.tabSize": 2,
```

Which files are you seeing with 4 spaces?

Comment (JimiC) [inline .editorconfig]: That darn `package.json`.

File: package.json (hunk 2 above)
Comment (robertohuertasm) [inline package.json]: I like it that way. It induces the idea that we're constantly evolving the extension.

File: .editorconfig (hunk 3 above)
Comment (robertohuertasm) [inline .editorconfig]: 😂 For what I see, most of the .json files. I'll change that.

File: package.json (hunk 2 above)
Comment (robertohuertasm) [inline package.json]: That's for sure! 😂

File: .editorconfig (hunk 3 above)
Comment (JimiC) [inline .editorconfig]: So are going with 2 or 4 spaces in `json` files?

Comment (robertohuertasm) [inline .editorconfig]: The idea is we go with 2

Comment (JimiC) [inline .editorconfig]: OK then, I'm done committing.

**Pull request merged**